### PR TITLE
Add support to publish the active session count

### LINKED ISSUE #9398: Active Session Count in IS analytics dashboard goes negative
IS Analytics Session count over time chart gets data from the 'SessionAggregation_<X>' (ex. SessionAggregation_DAYS) tables. Each of these table has 3 columns named as activeSessionCount, AGG_SUM_newSessionCount and AGG_SUM_terminatedSessionCount (see attachment). The value of the activeSessionCount column is calculated from a Siddhi app by adding AGG_SUM_newSessionCount to the previous activeSessionCount and reducing the AGG_SUM_terminatedSessionCount.

So in this case the negative activeSessionCount can occur when the AGG_SUM_terminatedSessionCount > previous(AGG_SUM_terminatedSessionCount) + AGG_SUM_newSessionCount. 

This might occur when the IS analytics started for the first time, there were active sessions which IS analytics didn't know. So the IS analytics assumes the activeSessionCount is 0 at that time and does the calculations based on the base value 0. But since there were active sessions, when those sessions are terminated, eventually the activeSessionCount values become negative.

**Suggested approaches to overcome the issue,**
1. If the activeSessionCount is negative, consider it as 0 in the database and other calculations.
2. Send the active session count from IS to IS analytics along with new and terminated counts.

### PR DESCRIPTION
**Pull request by pamodaaw**
Add support to publish the active session count as an attribute of the event stream for the analytics
Fixes wso2/product-is#9398

Depends on, https://github.com/wso2/carbon-identity-framework/pull/3116

### CODE REVIEW

File: components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASSessionDataPublisherImpl.java [hunk 1]
```diff
@@ -182,4 +165,81 @@ protected void publishSessionData(SessionData sessionData, int actionId) {
...
+        payloadData[8] = AuthPublisherConstants.NOT_AVAILABLE;
+        payloadData[9] = sessionData.getTenantDomain();
+        payloadData[10] = sessionData.getServiceProvider();
+        payloadData[11] = sessionData.getIdentityProviders();
+        payloadData[12] = sessionData.isRememberMe();
+        payloadData[13] = sessionData.getUserAgent();
+        payloadData[14] = System.currentTimeMillis();
+
+        if (LOG.isDebugEnabled()) {
+            for (int i = 0; i < 15; i++) {
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASSessionDataPublisherImpl.java]: Change 15 to payloadData.length

File: components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASSessionDataPublisherImpl.java [hunk 2]
```diff
@@ -182,4 +165,81 @@ protected void publishSessionData(SessionData sessionData, int actionId) {
...
+        payloadData[9] = sessionData.getTenantDomain();
+        payloadData[10] = sessionData.getServiceProvider();
+        payloadData[11] = sessionData.getIdentityProviders();
+        payloadData[12] = sessionData.isRememberMe();
+        payloadData[13] = sessionData.getUserAgent();
+        payloadData[14] = sessionData.getActiveSessionCount();
+        payloadData[15] = System.currentTimeMillis();
+
+        if (LOG.isDebugEnabled()) {
+            for (int i = 0; i < 16; i++) {
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASSessionDataPublisherImpl.java]: Change 16 to payloadData.length

Comment (pamodaaw) [reply components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASSessionDataPublisherImpl.java]: fixed

**Review by IsuraD — APPROVED** [review]

**Comment by pamodaaw** [comment]
The build: https://jenkins-support-all.wso2.com/jenkins/job/DEV_PR_BUILDER/907/, for the pull request: https://github.com/wso2-extensions/identity-data-publisher-authentication/pull/89, completed with the status: Successful

**Pull request merged**